system/jlib/jsocket.cpp

```diff
@@ -3623,6 +3615,7 @@ bool IpAddress::ipset(const char *text)
             return true;
     }
     memset(&netaddr,0,sizeof(netaddr));
+    hostname.clear();
```

Comment:
Not new, and not for now, but wondering why we always resolve 'netaddr' via lokupHostAddress for a host at this juncture.. when IpAddress is often just a container for a host/ip. It could avoid resolving it until needed.

Reply:
Still reviewing but yes exactly to this question.
If we are in ipset(char *) and its not an ipv4 or ipv6 dot-decimal string arg, then do we have to call:
        lookupHostAddress(text,netaddr);
to resolve it to an address right now ?
Only some routines actually need the netaddr, is there a way to delay lookup until those, unless serializing I suppose.

Reply:
Now it is a non-trivial class we could add a flag to indicate if the ip has been resolved.
I'm not sure how often it would help - how often are hostnames not converted to ip adresses?  Opened HPCC-30296 to track/discuss.